client/components/pie-chart/data/index.js

```diff
@@ -0,0 +1,27 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { isEqual, sortBy } from 'lodash';
+
+const NUM_COLOR_SECTIONS = 3;
+
+const assignSectionFunc = ( datum, index ) => {
+	return {
+		...datum,
+		sectionNum: index % NUM_COLOR_SECTIONS,
+	};
+};
+
+const sortDataAndAssignSections = data => {
+	return sortBy( data, datum => datum.value )
+		.reverse()
+		.map( assignSectionFunc );
+};
+
+const isDataEqual = ( currentData, newData ) => {
+	return isEqual( currentData, newData );
+};
```

Comment:
Why the dedicated helper function if the consuming component could continue to simply use `isEqual`? :smile:

Reply:
`PieChart` and `PieChartLegend` both consume an identical `data` prop. I wanted to make sure they both treat it the same. I admit the `isEqual` example is a little banal, but just in case there is a change made in the future both components will pick it up.

Reply:
I like to go with YAGNI in cases like this -- start with using the general-purpose `isEqual` in consumers, and only if the need arises, implement a specific helper.

(The situation -- deeply compare an array prop in a React lifecycle method to control its updating behavior -- seems common enough that I wouldn't expect this component to need anything more specific than `isEqual` anytime soon anyway :wink:)